contracts/contracts/Coordinator.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+/**
+* @title CoordinatorV1
+* @notice Coordination layer for DKG-TDec
+*/
+contract CoordinatorV1 {
+
+    uint8 public constant DKG_SIZE = 8;
+
+    event NewRitual(uint32 indexed ritualID, address[] nodes);
+    event TranscriptPosted(uint32 indexed ritualID, address indexed node, bytes32 transcriptDigest);
+    event ConfirmationPosted(uint32 indexed ritualID, address indexed node, address[] confirmedNodes);
```

Comment:
Is there any concern about having the list of nodes be publicly known and associated with a specific ritual ID?

Reply:
Yes, but that's still a hard problem, particularly if said nodes will post transcripts and confirmations to the blockchain anyway.